src/main/scala/com/greencatsoft/angularjs/extensions/FileUpload.scala

```diff
@@ -34,19 +34,25 @@ trait FileUploadConfig extends js.Object {
   var fileName: js.Any = js.native
 
   var fileFormDataName: js.Any = js.native
+  
+  var arrayKey: String = js.native
 }
 
 object FileUploadConfig {
 
-  def apply(url: String, file: File | js.Array[File], data: Option[Any] = None) = {
+  def apply(url: String, file: File | js.Array[File], data: Option[Any] = None, arrayKey: Option[String] = None) = {
     require(url != null, "Missing argument 'url'.")
     require(file != null, "Missing argument 'file'.")
 
     val config = new js.Object().asInstanceOf[FileUploadConfig]
 
     config.url = url
     config.file = file
-
+    
+    arrayKey match {
+      case Some(key) => {config.arrayKey = key}
```

Comment:
I think we better not use brackets here, as they are not needed and don't match the formatting convention.

By the way, wouldn't it throw a `MatchError` when the value of `arrayKey` is `None`? Maybe we can just write as follows:
```scala
arrayKey.foreach(config.arrayKey = _)
```